packages/techdocs-cli/src/commands/index.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommanderStatic } from "commander";
+
+export function registerCommands(program: CommanderStatic) {
+  program
+    .command("generate")
+    .description("Generate TechDocs documentation site using mkdocs.")
+    .option(
+      "--source-dir <PATH>",
+      "Source directory containing mkdocs.yml and docs/ directory.",
+      "."
+    )
+    .option(
+      "--output-dir <PATH>",
+      "Output directory containing generated TechDocs site.",
+      "./site/"
+    )
+    .option(
+      "--no-docker",
+      "Do not use Docker, use MkDocs executable and plugins in current user environment."
+    )
+    .option(
+      "--techdocs-ref <HOST_TYPE:URL>",
+      "The repository hosting documentation source files e.g. github:https://ghe.mycompany.net.com/org/repo." +
+        "\nThis value is same as the backstage.io/techdocs-ref annotation of the corresponding Backstage entity." +
+        "\nIt is completely fine to skip this as it is only being used to set repo_url in mkdocs.yml if not found.\n"
+    )
+    .option("-v --verbose", "Enable verbose output.", false)
+    .alias("build")
+    .action(lazy(() => import("./generate/generate").then(m => m.default)));
+
+  program
+    .command("publish")
+    .description(
+      "Publish generated TechDocs site to an external storage AWS S3, Google GCS, etc."
```

Comment:
I wonder if its needed to mention all different storage providers in the different descriptions of the commands, then you need to update here when there is a new storage provider supported.

I guess "Publish generated TechDocs site to an external storage." is enough? or?

Reply:
below you have the list of available type options anyways. Not super important to change, but just wanted to share my thinking.

Reply:
Thanks! I like this as of now since it helps in the description. But probably when the list grows large, we may not be able to write all of them at the top.